structuralcodes/codes/ec2_2023/_section_8_1_bending.py

```diff
@@ -0,0 +1,593 @@
+            confined concrete, then replace by fcd,c (8.15).
+        As (float): Reinforcement area in mm2.
+        fyd (float): Yield tensile resistance of steel in MPa.
+
+    Returns:
+        float: Axial resistance in compression in kN.
+
+    Raises:
+        ValueError: If any of Ac, fcd, As or Ayd is less than 0.
+    """
+    if Ac < 0:
+        raise ValueError('Concrete area cannot be negative')
+    if fcd < 0:
+        raise ValueError('Concrete resistance cannot be negative')
+    if As < 0:
+        raise ValueError('Steel area cannot be negative')
+    if fyd < 0:
+        raise ValueError('Steel resistance cannot be negative')
+
+    return (Ac * fcd + As * fyd) / 1000
+
+
+def biaxial_resistant_ratio(
+    MEdz_MRdz: float,
+    MEdy_MRdy: float,
+    Ned_NRd: float,
+    section_type: Literal['rectangular', 'circular', 'elliptical'],
+) -> float:
+    """Computes the resistant ratio in biaxial bending.
+
+    EN1992-1-1:2023 Eq. (8.2).
+
+    In the absence of an accurate cross-section design for biaxial beding this
+    criterion may be used.
+
+    Args:
+        MEdz_MRdz (float): Ratio between the design bending moment and the
+            resistance in the Z-axis.
+        MEdy_MRdy (float): Ratio between the design bending moment and the
+            resistance in the Y-axis.
+        Ned_NRd (float): Ratio between the design axial force and the axial
+            compressive resistance.
+        section_type (str): The section geometry type.
+
+    Returns:
+        float: The resistance ratio (non-dimensional).
+    """
+    if section_type in ('elliptical', 'circular'):
+        an = 2.0
+    elif Ned_NRd <= 0.1:
+        an = 1.0
+    elif Ned_NRd >= 1.0:
+        an = 2.0
+    else:
+        an = interp1d([0.1, 0.7, 2.0], [1.0, 1.5, 2.0])(Ned_NRd)
+
+    return abs(MEdz_MRdz) ** an + abs(MEdy_MRdy) ** an
+
+
+def sigma_cd(fcd: float, eps_c: float) -> float:
+    """Computes the stress distribution in the compression zones.
+
+    EN1992-1-1:2023 Eq. (8.4).
+
+    Computes the scress distribution in the cmpressiopn zones (compressive
+    shown as positive).
+
+    Args:
+        fcd (float): Compressive design resistance of concrete (MPa).
+        eps_c (float): Strain value of concrete (non dimensional).
+
+    Returns:
+        float: Concrete stress in MPa.
+
+    Raises:
+        ValueError: If strain greater than eps_c_u=0.0035.
+    """
+    if eps_c <= 0:
+        return 0.0
+    if eps_c <= 0.002:
+        return fcd * (1 - (1 - eps_c / 0.002) ** 2)
+    if eps_c <= 0.0035:
+        return fcd
+
+    raise ValueError('Strain cannot be greater than eps_c_u=0.0035')
+
+
+def delta_fcd_confined(sigma_c2d: float, f_cd: float, ddg: float) -> float:
+    """Calculate the compressive strength increase (delta_f_cd) due to a
+    transverse compressive stress.
+
+    EN1992-1-1:2023 Eq. (8.9 and 8.10).
+
+    Args:
+        sigma_c2d (float): Transverse compressive stress in MPa.
```

Comment:
should be this more clear?

`the absolute value of the minimum principal transverse compressive stress`